### PR TITLE
[keba] Ignore values if the curr hw commands sends 0

### PR DESCRIPTION
**Pull request by tiptick**
[KEBA] ignore 0 mA if keba booting up. 

on my p30c after bootup the "curr hw" is sometimes reporting 0 mA.
0 is a invalid value, only values from 6000 are valid. then it would set 6000 as min value,
but on first charging it will then not jump to 32000 but stay on 6000. as the documentation
tells "curr" is affecting the changes power output until keba is reset, or "currtime" is used.

tested on a P30c.
also the Documentation is stating values only from 6000 are valid.

### CODE REVIEW

File: bundles/org.openhab.binding.keba/src/main/java/org/openhab/binding/keba/internal/handler/KeContactHandler.java [hunk 1]
```diff
@@ -281,12 +281,17 @@ protected void onData(ByteBuffer byteBuffer) {
                         maxSystemCurrent = state;
                         State newState = new DecimalType(state);
                         updateState(new ChannelUID(getThing().getUID(), CHANNEL_MAX_SYSTEM_CURRENT), newState);
-                        if (maxSystemCurrent < maxPresetCurrent) {
-                            transceiver.send("curr " + String.valueOf(maxSystemCurrent), this);
-                            updateState(new ChannelUID(getThing().getUID(), CHANNEL_MAX_PRESET_CURRENT),
-                                    new DecimalType(maxSystemCurrent));
-                            updateState(new ChannelUID(getThing().getUID(), CHANNEL_MAX_PRESET_CURRENT_RANGE),
-                                    new PercentType((maxSystemCurrent - 6000) * 100 / (maxSystemCurrent - 6000)));
+                        if (maxSystemCurrent != 0){
```
Comment (kaikreuzer) [inline bundles/org.openhab.binding.keba/src/main/java/org/openhab/binding/keba/internal/handler/KeContactHandler.java]:
```suggestion
                        if (maxSystemCurrent != 0) {
```

File: bundles/org.openhab.binding.keba/src/main/java/org/openhab/binding/keba/internal/handler/KeContactHandler.java [hunk 2]
```diff
@@ -281,12 +281,17 @@ protected void onData(ByteBuffer byteBuffer) {
...
+                            if (maxSystemCurrent < maxPresetCurrent) {
+                                transceiver.send("curr " + String.valueOf(maxSystemCurrent), this);
+                                updateState(new ChannelUID(getThing().getUID(), CHANNEL_MAX_PRESET_CURRENT),
+                                        new DecimalType(maxSystemCurrent));
+                                updateState(new ChannelUID(getThing().getUID(), CHANNEL_MAX_PRESET_CURRENT_RANGE),
+                                        new PercentType((maxSystemCurrent - 6000) * 100 / (maxSystemCurrent - 6000)));
+                            }
+                        }
+                        else {
+                            logger.warn("maxSystemCurrent is 0. Ignoring.");
```
Comment (kaikreuzer) [inline bundles/org.openhab.binding.keba/src/main/java/org/openhab/binding/keba/internal/handler/KeContactHandler.java]:
```suggestion
                            logger.debug("maxSystemCurrent is 0. Ignoring.");
```

File: bundles/org.openhab.binding.keba/src/main/java/org/openhab/binding/keba/internal/handler/KeContactHandler.java [hunk 3]
```diff
@@ -281,12 +281,17 @@ protected void onData(ByteBuffer byteBuffer) {
...
-                                    new PercentType((maxSystemCurrent - 6000) * 100 / (maxSystemCurrent - 6000)));
+                        if (maxSystemCurrent != 0){
+                            if (maxSystemCurrent < maxPresetCurrent) {
+                                transceiver.send("curr " + String.valueOf(maxSystemCurrent), this);
+                                updateState(new ChannelUID(getThing().getUID(), CHANNEL_MAX_PRESET_CURRENT),
+                                        new DecimalType(maxSystemCurrent));
+                                updateState(new ChannelUID(getThing().getUID(), CHANNEL_MAX_PRESET_CURRENT_RANGE),
+                                        new PercentType((maxSystemCurrent - 6000) * 100 / (maxSystemCurrent - 6000)));
+                            }
+                        }
```
Comment (kaikreuzer) [inline bundles/org.openhab.binding.keba/src/main/java/org/openhab/binding/keba/internal/handler/KeContactHandler.java]: remove the line break

Comment (tiptick) [reply bundles/org.openhab.binding.keba/src/main/java/org/openhab/binding/keba/internal/handler/KeContactHandler.java]: sorry i did not get it, what linebreaks needs to be removed? from my view here it look good.

Comment (kaikreuzer) [reply bundles/org.openhab.binding.keba/src/main/java/org/openhab/binding/keba/internal/handler/KeContactHandler.java]: The else should be on the same line as the closing bracket:
```
} else {
```

**Comment by tiptick** [comment]
what is missing here?

**Comment by kaikreuzer** [comment]
Nothing, besides pinging me that you did the update (which I had missed). Thanks!

**Comment by tiptick** [comment]
thanks for merge ! appreciated !

**Pull request merged**